### PR TITLE
Task/wfh functionality and updating event service for future event types

### PR DESCRIPTION
**Pull request by Jamie-Unosquare**
Added Functionality and tests for working from home 
Added web mappings and removed duplicated web mapping
Made changes in the EventService and Interface to accept EventTypes to query for different types of event - also updated existing holiday code to reflect this change

### CODE REVIEW

File: v2/AdminCore.WebApp/Controllers/WorkingFromHomeController.cs [hunk 1]
```diff
@@ -12,37 +16,70 @@ namespace AdminCore.WebApi.Controllers
...
+      var eventDates = _mapper.Map<EventDateDto>(createWorkingFromHomeViewModel);
+
+      try
+      {
+        _eventService.CreateEvent(employeeId, eventDates, EventTypes.WorkingFromHome);
+        return Ok($"Work From Home has been created successfully");
+      }
+      catch (Exception e)
+      {
+        return Ok(e.Message);
```
Comment (nmcmahon-unosquare) [inline v2/AdminCore.WebApp/Controllers/WorkingFromHomeController.cs]: Return status code 500, log exception and return generic error message

**Review by nmcmahon-unosquare — CHANGES_REQUESTED** [review]

File: v2/AdminCore.WebApp/Controllers/WorkingFromHomeController.cs [hunk 2]
```diff
@@ -1,48 +1,89 @@
...
     public IActionResult GetWorkingFromHomeByEmployeeId(int employeeId)
     {
-      return Ok();
+      var wfhEvents = _eventService.GetEventsByEmployeeId(employeeId, EventTypes.WorkingFromHome);
+      if (wfhEvents != null)
+      {
+        return Ok(_mapper.Map<IList<WorkingFromHomeViewModel>>(wfhEvents));
+      }
+
+      return Ok($"No Work From Home found for employee ID: { employeeId.ToString() }");
```
Comment (BrownMark) [inline v2/AdminCore.WebApp/Controllers/WorkingFromHomeController.cs]: 204 not found here please.

**Comment by nmcmahon-unosquare** [comment]
Squash commits

**Review by eoinmcafee — APPROVED** [review]

File: v2/AdminCore.WebApp/Controllers/WorkingFromHomeController.cs (hunk 2 above)
Comment (eoinmcafee) [reply v2/AdminCore.WebApp/Controllers/WorkingFromHomeController.cs]: Can you address this one Jamie?

**Review by nmcmahon-unosquare — APPROVED** [review]

**Pull request merged**